### ISSUE TITLE
pack maker config request

### ISSUE THREAD

**Issue by wizard1399** [Leviathan143/betterbeginnings-MODIFIED#22]
Can you add a config to return the kiln to the same burn time as a vanilla furnace. I request this because a lot of packs now have some form of a mod that can make "mini coal" or "mini charcoal" that is suppose to burn exactly 1 item. Since the kiln has a different burn time, this causes difficulty.

**Comment by Leviathan143** [comment]
Yeah, sure

**Comment by belathus** [comment]
Also, can we have a separate config option for the obsidian kiln's burn time?

Looking through the code (and the wiki) the obsidian kiln used to be faster than the regular kiln. Right now the only purpose of the obsidian kiln is to be able to accept buckets of lava and blaze rods as fuel?

**Comment by Leviathan143** [comment]
This is not very high on the list of priorities, but I have been considering making the burn time of all the kilns configurable.

>Looking through the code (and the wiki) the obsidian kiln used to be faster than the regular kiln. Right now the only purpose of the obsidian kiln is to be able to accept buckets of lava and blaze rods as fuel?

Huh, looks like that got lost in one of the refactors. I'll fix that.

**Comment by belathus** [comment]
Would it be possible to add recipes that only work in the obsidian kiln? I'm thinking that this'll be an option for modpack devs.

**Comment by Leviathan143** [comment]
> Would it be possible to add recipes that only work in the obsidian kiln? I'm thinking that this'll be an option for modpack devs.

Yes. However it would be a lot of work, considering that I'd also need to do the same for the Nether Brick Oven and Ender Smelter as well. I'm going to need more than "it would be nice for modpack devs".

**Comment by Leviathan143** [comment]
@wizard1399 There's now Minetweaker support for changing burn times of fuels for BB machines. It's not in a released version of BB yet, but does it solve this issue?

**Comment by wizard1399** [comment]
I will look into it to check when I can next fire up MC.

**Comment by wizard1399** [comment]
Can not find the documentation for this change. It might be better to make the fuel multiplier a variable that is set in config. If I had to guess, the current code is checking the burn time of the fuel inserted and multiplying it by a number less than 1.0 to get the current result.

**Comment by Leviathan143** [comment]
@wizard1399 
>Can not find the documentation for this change. 

It's not released yet, so it isn't documented. 

> It might be better to make the fuel multiplier a variable that is set in config. If I had to guess, the current code is checking the burn time of the fuel inserted and multiplying it by a number less than 1.0 to get the current result.

Burn time of an item is determined by first checking a map of overrides that refer to specific items with specific nbt and specific metadata. If there is no applicable override, a fallback fuel handler determines the burn time depending on Material or other properties. All fuel consumers have their own override map and fallback handler, except for the NB oven which uses separate logic due to using fluid fuel. 

The MT support allows you to specify the burn time in ticks of a specific item in a specific fuel consumer. An example line of ZenScript would be `FuelTweaker.setFuelVale("CAMPFIRE", <minecraft:rotten_flesh>, 100)`, which would make Rotten Flesh burn for 100 ticks in the campfire. Sorry for not specifying this. Does the MT support solve this issue?

**Comment by wizard1399** [comment]
Ahh, I see. Did not realise that it was a lookup map method for this. That does mean more work for pack makers as they will have to find and adjust every fuel source in the pack. Could a variable rate be multiplied by entries in the lookup table? In theory this would adjust all fuel types. For example: look up coal and get burn time; multiply this by variable set in config; return result as burn time. Would that work?

**Comment by wizard1399** [comment]
Does the zs above accept ore dict entries as the fuel input type?

**Comment by Leviathan143** [comment]
> Ahh, I see. Did not realise that it was a lookup map method for this. That does mean more work for pack makers as they will have to find and adjust every fuel source in the pack. Could a variable rate be multiplied by entries in the lookup table? In theory this would adjust all fuel types. For example: look up coal and get burn time; multiply this by variable set in config; return result as burn time. Would that work?

Yes, it would be easy to implement. However I expect it would also cause confusion when the actual burn time of a fuel doesn't match the set burn time due to the multiplier.

>Does the zs above accept ore dict entries as the fuel input type?

No. It does support wildcard damage values though. It shouldn't be hard to implement that though.
